### PR TITLE
Deprecated GET /branch/{branchid}

### LINKED ISSUE #74: Deprecate endpoint GET /branch/{branchid}
> Based on [RFC-0016](https://iver-wharf.github.io/rfcs/published/0016-wharf-api-endpoints-cleanup) (https://github.com/iver-wharf/rfcs/pull/16)

See note from RFC:

> Deprecated. Has not been moved, but instead planned to be removed. Was not implemented in v4, and its usage is replaced instead by the `GET /project/{projectId}/branch` endpoint.
>
> https://iver-wharf.github.io/rfcs/published/0016-wharf-api-endpoints-cleanup#get-branchbranchid

### PR DESCRIPTION
**Pull request by jilleJr**
- \[x] I've added a new note in the `CHANGELOG.md` file, according to docs:
  https://iver-wharf.github.io/#/development/changelogs/writing-changelogs

## Summary

- Moved handler for `GET /branch/{branchid}` into new package `internal/deprecated`
- Added `@deprecated` flag to swaggo comment

## Motivation

Closes #74

There were other endpoints planned for deprecation, such as `POST /project/search`, but I'm leaving that for the PR of #71.

The other "Not implemented" endpoints in the `build.go` file are planned for adding implementation to in https://github.com/iver-wharf/wharf-api/milestone/1

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,10 @@ This project tries to follow [SemVer 2.0.0](https://semver.org/).
   after they were initially used, leading to a mix of wharf-core and Gin
   formatted logs. (#63)
 
+- Deprecated endpoint `GET /branch/{branchid}`. Getting a single branch by its
+  ID has been not shown any benefits. Please refer to the
+  `GET /project/{projectid}` endpoint instead. (#75)
```
Comment (Alexamakans) [inline CHANGELOG.md:26]:
```suggestion
- Deprecated endpoint `GET /branch/{branchid}`. Getting a single branch by its
  ID has not been shown to have any benefits. Please refer to the
  `GET /project/{projectid}` endpoint instead. (#75)
```

**Review by Alexamakans — APPROVED** [review]

**Review by iverestefans — APPROVED** [review]
LGTM!

**Review by fredx30 — COMMENTED** [review]
Why are we moving this? 
For a deprecated method/route/api i would expect removal or an ignore.

**Comment by jilleJr** [comment]
> Why are we moving this?
> For a deprecated method/route/api i would expect removal or an ignore.

It was moved to `internal/deprecated` package to get it out of the way. First we mark it as deprecated, and then in a later version bump we remove it.

**Pull request merged**